src/classes/OPP_OpportunityAmountSync_SVC.cls

```diff
@@ -0,0 +1,32 @@
+public class OPP_OpportunityAmountSync_SVC {
+
+    /**
+     * Put the fields needed in the query in the description.
+     * @param opportunitiesByIds
+     * @return
+     */
+    public List<Opportunity> syncOppAmountWithTotalPaidPaymentAmount(Map<Id, Opportunity> opportunitiesByIds) {
+        List<Opportunity> oppsToUpdate = new List<Opportunity>();
+        
+        for (Opportunity opp : opportunitiesByIds.values()) {
+            Decimal totalPaid = 0;
+
+            for (npe01__OppPayment__c pmt : opp.npe01__OppPayment__r) {
+                if (pmt.npe01__Paid__c) {
+                    totalPaid += pmt.npe01__Payment_Amount__c == null ? 0 : pmt.npe01__Payment_Amount__c;
+                }
+            }
+
+            Boolean addOppToList = (OPP_StageMappingUtil.determineOppStateFromOppStage(opp.StageName) == 'FINALIZED' && opp.Amount != totalPaid) ||
```

Comment:
I asked @KietTruongSFDO in his related PR to make a method in OPP_StageMappingUtil called either isFinalized or isOpportunityFinalized, depending on whether the Opportunity or the stage is passed (I originally indicated I don't care which, but in Abacus when we did this, we also checked the Opportunities isClosed field, and if it was not a Closed stage, we threw an exception.  I believe we should do the same here [though I could be convinced we should do this later if we are trying to get it in today]).  Can you work with him and agree on that, and get this in there?